### PR TITLE
[Search] Fix nightly and weekly test failure

### LINKED ISSUE #33356: [Search][BUG] Fix "options" test failure in weekly and nightly runs
### Library name

Azure.Search.Documents

### Please describe the feature.

image

Weekly pipeline failure - https://dev.azure.com/azure-sdk/internal/_build/results?buildId=2099430&view=logs&j=d76cd559-fb8e-56d0-cec0-e45f7cdf7dbf&t=be861a6a-e72b-540a-ce94-c251e2053297&l=1272

Nightly pipeline failure - https://dev.azure.com/azure-sdk/internal/_build/results?buildId=2099511&view=logs&j=32e69777-6632-566b-8772-98de52ef067a&t=d8987eda-cbe3-5fa2-c061-f64e97907b47&l=528

### PR DESCRIPTION
**Pull request by ShivangiReja**
Fixes: https://github.com/Azure/azure-sdk-for-net/issues/33356

### CODE REVIEW

File: sdk/search/Azure.Search.Documents/tests/Utilities/SearchResources.cs [hunk 1]
```diff
@@ -455,40 +455,58 @@ private async Task<SearchResources> CreateHotelsBlobContainerAsync(bool populate
...
-
-                    foreach (object obj in hotels)
+                    if (isSample)
                     {
-                        Task task = Task.Run(async () =>
+                        Samples.Hotel[] hotels = SearchResourcesSample.TestDocumentsForSample;
+                        List<Task> tasks = new List<Task>(hotels.Length);
+
+                        foreach (Samples.Hotel hotel in hotels)
                         {
```
Comment (pallavit) [inline sdk/search/Azure.Search.Documents/tests/Utilities/SearchResources.cs:464]: [nit] You could move the common code like tasks and waiting on them outside of if\else and share it.

**Review by pallavit — APPROVED** [review]
:shipit:

File: sdk/search/Azure.Search.Documents/README.md [hunk 2]
```diff
@@ -317,7 +319,7 @@ SearchOptions options = new SearchOptions
     // Filter to only Rating greater than or equal our preference
     Filter = SearchFilter.Create($"Rating ge {stars}"),
     Size = 5, // Take only 5 results
-    OrderBy = { "Rating desc" } // Sort by Rating from high to low
+   // OrderBy = { "Rating desc" } // Sort by Rating from high to low
```
Comment (heaths) [inline sdk/search/Azure.Search.Documents/README.md:322]: Given this shows up in a sample, I'd remove this line entirely. In a sample, it seems confusing.

**Review by heaths — APPROVED** [review]
Minor comment, but otherwise LGTM.

File: sdk/search/Azure.Search.Documents/README.md (hunk 2 above)
Comment (ShivangiReja) [reply sdk/search/Azure.Search.Documents/README.md]: [This](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=2101992&view=logs&j=283d73df-0aa6-5cae-0d63-20b601dc19a3&t=40ae92eb-816f-5cde-3fce-f7b41ef072b5&l=562) test case is passing locally(in live, record and playback all three modes) but failing in nightly pipeline. So I was just trying to debug and commented out this line for testing purposes. 

This line is now uncommented.

File: sdk/search/Azure.Search.Documents/README.md [hunk 3]
```diff
@@ -319,7 +319,7 @@ SearchOptions options = new SearchOptions
     // Filter to only Rating greater than or equal our preference
     Filter = SearchFilter.Create($"Rating ge {stars}"),
     Size = 5, // Take only 5 results
-   // OrderBy = { "Rating desc" } // Sort by Rating from high to low
+    OrderBy = { "Rating desc" } // Sort by Rating from high to low
```
Comment (heaths) [inline sdk/search/Azure.Search.Documents/README.md:322]: If you're adding this back in, shouldn't you re-record? That said, as long as you're not asserting the order, I guess the already-recorded session doesn't really matter.

Comment (ShivangiReja) [reply sdk/search/Azure.Search.Documents/README.md:322]: I didn't change the recordings when I commented out this line for nightly testing purposes. 

[Here's](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=2109985&view=logs&j=d76cd559-fb8e-56d0-cec0-e45f7cdf7dbf&t=be861a6a-e72b-540a-ce94-c251e2053297) the recording for OrderBy.

File: sdk/search/Azure.Search.Documents/tests/Samples/Sample01_HelloWorld.cs [hunk 4]
```diff
@@ -49,6 +49,8 @@ public async Task CreateClient()
             Response<SearchServiceStatistics> stats = indexClient.GetServiceStatistics();
             Console.WriteLine($"You are using {stats.Value.Counters.IndexCounter.Usage} indexes.");
             #endregion Snippet:Azure_Search_Tests_Samples_CreateClient
+
+            Assert.GreaterOrEqual(stats.Value.Counters.IndexCounter.Usage, 1);
```
Comment (pallavit) [inline sdk/search/Azure.Search.Documents/tests/Samples/Sample01_HelloWorld.cs:53]: Is this timing issue? Curious how this resulted in tests failure when run all together vs not!

Comment (ShivangiReja) [reply sdk/search/Azure.Search.Documents/tests/Samples/Sample01_HelloWorld.cs:53]: No, this is a different test and yes, it's timing issue for this test case.

Previously [Options](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=2109985&view=logs&j=32e69777-6632-566b-8772-98de52ef067a&t=d8987eda-cbe3-5fa2-c061-f64e97907b47&l=528) test case was failing. The issue was the common [EnsureInitialized](https://github.com/ShivangiReja/azure-sdk-for-net/blob/52ab5e3dd1a9b553f6aea85b74c4d6b6ea068dc7/sdk/search/Azure.Search.Documents/tests/Utilities/SharedSearchResources.cs#L31) method, which we were using for sample and test both. Now, we use TestDocumentsForSample(uploads document in pascal case like public portal) for samples and TestDocument(upload document in camel case) for tests (see [here](https://github.com/ShivangiReja/azure-sdk-for-net/blob/52ab5e3dd1a9b553f6aea85b74c4d6b6ea068dc7/sdk/search/Azure.Search.Documents/tests/Utilities/SearchResources.cs#L424-L431)). [EnsureInitialized](https://github.com/ShivangiReja/azure-sdk-for-net/blob/52ab5e3dd1a9b553f6aea85b74c4d6b6ea068dc7/sdk/search/Azure.Search.Documents/tests/Utilities/SearchResources.cs#L273) method was initializing for tests and using the same test resources for both tests and samples, so some of samples were failing. [Here](https://github.com/Azure/azure-sdk-for-net/pull/33364/files#diff-b9e83f9dcb19a960cc355b5ed62e8b3337d10446f1d86f41ce4fb3fc96d55a3c) I added the fix for it. 

Now all the test cases are passing and all the pipelines are green -- https://dev.azure.com/azure-sdk/internal/_build/results?buildId=2115015&view=results.

**Pull request merged**